### ISSUE TITLE
create an internals documentation folder in the cpython repo

### ISSUE THREAD

**Issue by iritkatriel** [python/cpython#119786]
The internals documentation is scattered in markdown files in the codebase, as well as parts which are in the dev guide.  We would like to have it in one place, versioned along with the code (unlike the dev guide). 

### Linked PRs
* gh-119787
* gh-119815
* gh-120137
* gh-120134
* gh-120445
* gh-121009
* gh-123198
* gh-124450
* gh-124989
* gh-125119
* gh-125282
* gh-125715
* gh-125874
* gh-125888
* gh-126832
* gh-127175
* gh-128329
* gh-128524
* gh-131203
* gh-131213
* gh-131382
* gh-135411
* gh-141816
* gh-142812

### Cosmetic PRs

* gh-120077
* gh-121601
* gh-124990
* gh-125455
* gh-125456
* gh-125815
* gh-127485
* gh-127957
* gh-128174
* gh-128275
* gh-128314

**Comment by colesbury** [comment]
FYI, there are a few broken links:

* `frame_layout.md`: https://github.com/search?q=repo%3Apython%2Fcpython%20frame_layout.md&type=code
* The first few links in this section: https://github.com/python/cpython/blob/main/InternalDocs/compiler.md#objects

**Comment by iritkatriel** [comment]
Yes, but it needs to be reviewed to make sure verrything is up to date while we do that. Would you like to make a PR? I'm happy to help review.

**Comment by Eclips4** [comment]
> Yes, but it needs to be reviewed to make sure verrything is up to date while we do that. Would you like to make a PR? I'm happy to help review.

Yes, I'd be happy to send a PR.
